app/assets/javascripts/discourse/app/models/category.js

```diff
@@ -485,6 +485,16 @@ Category.reopenClass({
     return ajax(`/c/${slugPath}/find_by_slug.json`);
   },
 
+  reloadCategoryWithPermissions(params, store, site) {
+    return this.reloadBySlug(params.slug, params.parentSlug).then((result) => {
+      const record = store.createRecord("category", result.category);
+      record.setupGroupsAndPermissions();
+      record.set("params", params);
```

Comment:
It looks like you're params to the category itself, which is a bit odd. It means that due to our identity map those params are going to follow those models around everywhere. It's also not a great separation of concerns as the params to find a category don't belong to the category object itself.

Instead the `params` should be set on a controller if you need them. It seems like the params are passed through so that you can dynamically build the link? Could you instead pass the `category` object into the component so that the slug and parentslug can be queried via that object?

Reply:
Yes, that was awkward, very appropriate. I have modified this so the `params` are sent to the controller (and then component) from each route (and there are three routes using this: parent category, child category and new category) outside the model. It made more sense to me than passing the category object to component, that seemed like overkill (don't hesitate to correct me if this is wrong).

I also renamed the main route/controller to `edit-category-tabs` to avoid confusion with each individual `edit-category-tab` component.